rb/lib/selenium/webdriver/common/credential.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#
+# A credential stored in a virtual authenticator.
+# @see https://w3c.github.io/webauthn/#credential-parameters
+#
+
+module Selenium
+  module WebDriver
+    class Credential
+      def initialize(*args)
```

Comment:
Rather than using positional arguments, let's use keyword arguments:
```suggestion
      def initialize(id:, is_resident_credential:, rp_id:, user_handle:, private_key:, sign_count:)
```

If any of these parameters is optional, we can just set the default value to `nil`.

Reply:
I have made the arguments as positional. But in context of default values there's a doubt.
Required parameters are : `id, rp_id, private_key, sign_count` 

The remaining two parameter i.e `is_resident_credential` and `user_handle` are used in combination. For example,
When `is_resident_credential` = true, `user_handle` != nil
When `is_resident_credential` = true, `user_handle` == nil

I don't suggest making either of them optional as case may arise when user puts `is_resident_credential` as true but doesn't assign `user_handle`.

Reply:
In this case, I would make only `id, rp_id, private_key, sign_count` required and then add additional validations for arguments that are connected:

```ruby
msg = "Both is_resident_credential and user_handle should be passed together"
raise ArgumentError, msg if is_resident_credential && user_handle.nil?
raise ArgumentError, msg if !is_resident_credential && !user_handle.nil?
```